components/pedigree/resources/src/main/resources/pedigree/view/tabbedSelector.js

```diff
@@ -98,18 +98,25 @@ define([
 
             this.dialog.show();
 
-            if (!allowCancel) {
-                // hide close button and disable keyboard close shortcut
-                $(this.dialog.getBoxId()).down('.msdialog-close').hide();
-                this.dialog.unregisterShortcuts("close");
-            } else {
-                // show close button and enable keyboard close shortcut
-                $(this.dialog.getBoxId()).down('.msdialog-close').show();
-                this.dialog.registerShortcuts("close");
+            var closeButton = this.dialog.dialog.down('.msdialog-close');
+            var closeFunction = this.hide.bindAsEventListener(this);
+            if (!stayInEditorOnClose) {
+                closeFunction = function(event) {
+                    event.stop();
+                    new Ajax.Request(XWiki.currentDocument.getURL('delete', 'confirm=1&ajax=1&form_token=' + $$('meta[name="form_token"]')[0].content), {
+                        'method': 'GET',
+                        'onSuccess': function() { history.go(-1); }
```

Comment:
...arguably `history.go(-1)` can be done even if family removal fails? Possibly after a dialog warning about that?